src/Hamiltonians/ReducedDensityMatrix.jl

```diff
@@ -0,0 +1,150 @@
+"""
+    SingleParticleReducedDensityMatrix(i, j) <: AbstractHamiltonian
+
+Represent a {i,j} element of the single-particle reduced density matrix:
+
+```math
+\\hat{ρ}^{(1)}_{i,j} = \\langle \\psi | \\hat a^†_{i} \\hat a_{j} | \\psi \\rangle
+```
+
+where ``i`` and ``j`` are the `mode` and ``| \\psi \\rangle`` is the `state-ket` of the given Hamiltonian.
+
+# See also
+
+* [`single_particle_density`](@ref)
+* [`SingleParticleDensity`](@ref)
+* [`TwoParticleReducedDensityMatrix`](@ref)
+"""
+struct SingleParticleReducedDensityMatrix <: AbstractHamiltonian{Float64}
+    I::Int
+    J::Int
+end
+
+function Base.show(io::IO, spd::SingleParticleReducedDensityMatrix)
+    print(io, "SingleParticleReducedDensityMatrix($(spd.I), $(spd.J))")
+end
+
+LOStructure(::Type{T}) where T<:SingleParticleReducedDensityMatrix = AdjointUnknown()
+
+function diagonal_element(spd::SingleParticleReducedDensityMatrix, add::SingleComponentFockAddress)
+    src = find_mode(add, spd.J)
+    dst = find_mode(add,spd.I)
+    address, value = excitation(add, (dst,), (src,))
+    if spd.I == spd.J
+        return value
+    else
+        return 0.0
+    end
```

Comment:
```suggestion
    if spd.I != spd.J
        return 0.0
    end
    src = find_mode(add, spd.J)
    return src.occnum
```
For performance reasons its better to first check the equality and then avoid doing unnecessary calculations.